src/shared/udev-util.c

```diff
@@ -159,7 +160,8 @@ static int device_wait_for_initialization_internal(
         _cleanup_(sd_device_monitor_unrefp) sd_device_monitor *monitor = NULL;
         _cleanup_(sd_event_source_unrefp) sd_event_source *timeout_source = NULL;
         _cleanup_(sd_event_unrefp) sd_event *event = NULL;
-        struct DeviceMonitorData data = { .devlink = devlink };
+        _cleanup_(sd_device_unrefp) sd_device *ret_device = NULL;
+        struct DeviceMonitorData data = { .devlink = devlink, .device = &ret_device };
```

Comment:
hmm, i would just do it like this:

```c
void device_monitor_data_free(DeviceMonitoData *d) {
        sd_device_unref(d->device);
}

static int device_wait_for_initialization_internal(
        …
        _cleanup_(device_monitor_data_free) struct DEviceMonitorData data = …
```

that's a bit more future proof I think, and easier on the eyes, given that there are fewer pointer redirections... And it's a shorter patch too

Reply:
Ok, done